tests/discovery/Discovery.test.ts

```diff
@@ -152,7 +162,6 @@ describe('Discovery', () => {
     });
     await nodeManager.start();
     await nodeConnectionManager.start({ nodeManager });
-    await taskManager.startProcessing();
     // Set up other gestalt
     nodeA = await PolykeyAgent.createPolykeyAgent({
```

Comment:
We should abstract this creation of PolykeyAgents for testing purposes. There's a lot of tests that are repeating this bootstrapping code. I thought we had something like this already called `pkAgent`, but I think that was for spawning a separate process. We should have something similar for starting in-process agents.

Reply:
It was called `setupGlobalAgent`, but it was removed since we no longer required a global agent process. But it does seem having a utility for in-process agents would be helpful here.

Reply:
Actually I was confused about this, it's the discovery dependencies themselves.

Reply:
I can but the change will have to be applied to a broad range of tests. This would be out of scope for the PR. I can make an issue for it.

Reply:
Not important atm.